framework/doc/content/source/actions/Action.md

```diff
@@ -1,20 +1,108 @@
-An action is assocated with tasks, which determine when the action is to be acted, or the *act* function of an action is to be called by MOOSE.
-Association with multiple tasks is allowed. In such case, it is developers' responsibility to switch the functions based on the current task in the act function.
-A full list of tasks is available in Moose.C file.
-Developers are allowed to register new tasks with MOOSE Action system.
-When actions are built either through the input parser or internally by MOOSE, they will exist during the entire simulation.
-
-MOOSE has lots of built-in actions for adding individual objects through input files.
-But MOOSE action system is far beyond the basic capabilities provided by these built-in actions.
-Developers are encouraged to explore the MOOSE actions system to create their own actions in order to perform problem setup in one place and potentially simplify the input syntax significantly.
-One example is to use an action to append a mesh generator that generates a
-boundary surrounding a particular subdomain for imposing specific boundary
-conditions, based on the information not necessarily inside of a mesh generator
-block.
+MOOSE *Actions*  and are used to execute *tasks*. Each application registers
+numerous *actions*, *tasks*, and *syntax*. Each task is associated with one or more
+actions, and each action may perform one or more tasks. Syntax is used by the
+input file parser to generate actions/tasks.
+
+Common uses for actions are to perform setup tasks and create MOOSE objects.
+
+## Creating Actions
+
+To create a new action, first derive from the appropriate base class: if the new action
+is to correspond to creating MOOSE objects from an input file, then derive
+from `MooseObjectAction`; else, derive from `Action`.
+
+The `act()` method must be implemented to perform the associated task(s). If the
+action will be registered to multiple tasks, then the variable `_current_task`
+can be queried to determine the current task, for example,
+
+``` language=cpp
+void
+ExampleAction::act()
+{
+  if (_current_task == "example_task_a")
+  {
+    // "example_task_a" execution
+  }
+  else if (_current_task == "example_task_b")
+  {
+    // "example_task_b" execution
+  }
+}
+```
+
+`MooseObjectAction`s, have the member variables `_type` and `_moose_object_pars`,
+which correspond to the type and [InputParameters.md] of the MOOSE object to
+be created, respectively. For example, the action to create a BC object has
+the following `act()` method:
+
+!listing framework/src/actions/AddBCAction.C re=void\sAddBCAction::act.*?^}
+
+The action should be registered to one or more tasks using the `registerMooseAction`
+macro (conventionally in the action source file), for example,
+
+``` language=cpp
+registerMooseAction("ExampleApp", ExampleAction, "example_task_a");
+registerMooseAction("ExampleApp", ExampleAction, "example_task_b");
+```
+
```

Comment:
Suggest to introduce `registerSyntax` here as well. This can also a good place for introducing auto-built actions there do not require input syntax.